transformer_engine/pytorch/csrc/extensions/attention.cu

```diff
@@ -510,6 +511,102 @@ std::vector<py::object> fused_attn_bwd(
   return {py_dQ, py_dK, py_dV, py::cast(dBias)};
 }
 
+/***************************************************************************************************
+ * Support BSHD, SBHD, and THD formats for Context Parallel: Fused out correction in forward
+ **************************************************************************************************/
+
+template <typename dtype, bool causal>
+void fused_out_correction_helper(at::Tensor &out, const std::vector<at::Tensor> &out_per_step,
+                                 const at::Tensor &lse, const std::vector<at::Tensor> &lse_per_step,
+                                 const py::object &cu_seqlens, std::string qkv_format, int cp_size,
+                                 int rank, bool softmax_lse_in_packed_format) {
+  std::unordered_map<std::string, NVTE_QKV_Format> format_map = {
+      {"sbhd", NVTE_QKV_Format::NVTE_SBHD},
+      {"bshd", NVTE_QKV_Format::NVTE_BSHD},
+      {"thd", NVTE_QKV_Format::NVTE_THD}};
```

Comment:
this looks weird, you should change `std::string qkv_format` to `NVTE_QKV_Format qkv_format`.

in `attention.py`, do something like `import QKVFormat` from cpp_extensions/fused_attn, then pass `QKVFormat[qkv_format]` to `tex.fused_out_correction`